helpers/validator.js

```diff
@@ -0,0 +1,127 @@
+const steem = require('@steemit/steem-js');
+const validator = require('validator');
+const badDomains = require('../bad-domains');
+
+const accountNotExist = (rule, value, callback) => {
+    steem.api.getAccounts([value], (err, result) => {
+        if (result[0]) {
+            callback(['Account name is not available']);
+        } else {
+            callback();
+        }
+    });
+};
+
+const INVALID_ACCOUNTNAME_REASONS = {
+    error_username_required: 'error_username_required',
+    error_validation_account_min: 'error_validation_account_min',
+    error_validation_account_max: 'error_validation_account_max',
+    error_validation_account_start: 'error_validation_account_start',
+    error_validation_account_segment_start:
+        'error_validation_account_segment_start',
+    error_validation_account_alpha: 'error_validation_account_alpha',
+    error_validation_account_segment_alpha:
+        'error_validation_account_segment_alpha',
+    error_validation_account_dash: 'error_validation_account_dash',
+    error_validation_account_segment_dash:
+        'error_validation_account_segment_dash',
+    error_validation_account_end: 'error_validation_account_end',
+    error_validation_account_segment_end:
+        'error_validation_account_segment_end',
+    error_validation_account_segment_min:
+        'error_validation_account_segment_min',
+};
+
+/**
+ * If the name is invalid, emits one of INVALID_ACCOUNTNAME_REASONS.
+ *
+ * @param {string} name
+ * @return {string|false}
+ */
+// https://github.com/steemit/condenser/blob/eaf8a02658b8deaef376ec90b81d0866e52582cc/app/utils/ChainValidation.js#L4
+const accountNameIsInvalid = name => {
+    let i;
+    let label;
+    let len;
+
+    if (!name) {
+        return INVALID_ACCOUNTNAME_REASONS.error_username_required;
```

Comment:
Throwing an error instead of mixing boolean and string return types would be a cleaner implementation